packages/fuels-core/src/types/core/bytes.rs

```diff
@@ -1,6 +1,29 @@
+use crate::types::errors::{error, Result};
+
 #[derive(Debug, PartialEq, Clone, Eq)]
 pub struct Bytes(pub Vec<u8>);
 
+impl Bytes {
+    /// Create a new `Bytes` from a string representation of a hex.
+    /// Accepts both `0x` prefixed and non-prefixed hex strings.
+    pub fn from_hex_str(hex: &str) -> Result<Self> {
+        let hex = if let Some(stripped_hex) = hex.strip_prefix("0x") {
+            stripped_hex
+        } else {
+            hex
+        };
```

Comment:
```suggestion
        let hex = hex.strip_prefix("0x").unwrap_or(hex);
```

Reply:
A release is coming up and it would be great if this was part of it. Since we need to do it now, we'll merge without this suggestion and apply it afterward. 

@Br1ght0ne